### PR TITLE
[12.0][MIG+REFACTOR] account_taxjar, sale_account_taxjar, base_taxjar

### PR DESCRIPTION
**Pull request by HviorForgeFlow**
CC ~  @Eficent @mreficent @jbeficent @lreficent @aheficent @ageficent

### CODE REVIEW

File: base_taxjar/README.rst [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+   :align: center
+
+** You must configure manually Sourcing Type...
+`read more <https://blog.taxjar.com/charging-sales-tax-rates/>`_
+
+Known issues / Roadmap
+======================
+
+- [x] Consider Logistic Routes to determine from address jurisditions
+- [x] Move TaxJar Configuration to Settings.
```
Comment (LoisRForgeFlow) [inline base_taxjar/README.rst]: If done, then remove them

File: base_taxjar/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html).
+{
+    'name': 'Base TaxJar',
+    'version': '11.0.1.0.0',
```
Comment (LoisRForgeFlow) [inline base_taxjar/__manifest__.py]: 12

File: sale_account_taxjar/tests/test_account_taxjar.py [hunk 3]
```diff
@@ -0,0 +1,171 @@
+# Copyright 2018-2019 Eficent Business and IT Consulting Services S.L.
+# - (https://www.eficent.com)
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl.html).
+import logging
+from os.path import dirname, join
+from odoo.tests import SingleTransactionCase
+from .utils import scrub_string
+
+from vcr import VCR
```
Comment (LoisRForgeFlow) [inline sale_account_taxjar/tests/test_account_taxjar.py]: Cool :+1:

**Review by LoisRForgeFlow — COMMENTED** [review]
some small comments

File: account_taxjar/migrations/12.0.1.0.0/post-migration.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+            'name': name,
+            'taxjar_id': taxjar_id,
+            'sourcing_type': sourcing_type,
+            'taxable_account_id': taxable_account_id,
+        })
+        afp = account_fiscal_position.browse(afp_id)
+        afp.write({'taxjar_nexus_sourcing_id': tns.id})
+
+
+def migrate(cr, version):
```
Comment (MiquelRForgeFlow) [inline account_taxjar/migrations/12.0.1.0.0/post-migration.py]: why not this?
```suggestion
@openupgrade.migrate()
def migrate(env, version):
```

File: account_taxjar/migrations/12.0.1.0.0/pre-migration.py [hunk 5]
```diff
@@ -0,0 +1,23 @@
+# Copyright 2019 Eficent Business and IT Consulting Services S.L.
+#   (http://www.eficent.com)
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from openupgradelib import openupgrade
+
+_model_renames = [
+    ('product.taxjar.category', 'taxjar.category'),
+    ('base.account.taxjar', 'taxjar.api')
```
Comment (MiquelRForgeFlow) [inline account_taxjar/migrations/12.0.1.0.0/pre-migration.py]: problem: when executing this, `base_taxjar` is already installed.

File: account_taxjar/migrations/12.0.1.0.0/post-migration.py [hunk 6]
```diff
@@ -0,0 +1,37 @@
+# Copyright 2019 Eficent Business and IT Consulting Services, S.L.
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl.html).
+import logging
+from openupgradelib import openupgrade
+
+_logger = logging.getLogger(__name__)
+
+__name__ = "Upgrade to 11.0.2.0.0"
```
Comment (MiquelRForgeFlow) [inline account_taxjar/migrations/12.0.1.0.0/post-migration.py]: wrong, delete this and the logger, no needed

File: account_taxjar/views/account_fiscal_position.xml [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+            <field name="name">account.fiscal.position.inherit.form</field>
+            <field name="model">account.fiscal.position</field>
+            <field name="inherit_id" ref="account.view_account_position_form"/>
+            <field name="arch" type="xml">
+                <field name="name" position="after">
+                    <field name="taxjar_nexus_sourcing_id"/>
+                </field>
+            </field>
+    </record>
+</odoo>
```
Comment (MiquelRForgeFlow) [inline account_taxjar/views/account_fiscal_position.xml]: add blank line

**Review by MiquelRForgeFlow — APPROVED** [review]
very good :+1:

**Pull request merged**